src/components/MovieList.js

```diff
@@ -2,44 +2,72 @@ import React from 'react';
 
 import MovieListItem from './MovieListItem';
 
+import {SORTING_BY_RATING, SORTING_BY_DATE} from "../constants";
+
 export default class MovieList extends React.PureComponent {
 
     constructor(props) {
         super(props);
 
         this.state = {
-            isEmpty: !(this.props.movieList && this.props.movieList.length)
+            isEmpty: !(this.props.movieList && this.props.movieList.length),
+            movieList: []
         }
     }
 
-
     static getDerivedStateFromProps(nextProps) {
         return {
+            movieList: MovieList.sortList(nextProps.sortingType, nextProps.movieList),
             isEmpty: !(
                 nextProps.movieList && nextProps.movieList.length
             )
         }
     }
 
+    static compareDates = (movie_1, movie_2) => {
+        const date_1 = String(movie_1.year).split('-').join('');
+        const date_2 = String(movie_2.year).split('-').join('');
+
+        return Number(date_2) - Number(date_1);
+    };
+
+    static compareRating = (movie_1, movie_2) => (
+        Number(movie_2.rating) - Number(movie_1.rating)
+    );
+
+    static sortList(sortingType, list) {
+        list = [...list];
```

Comment:
reassigning function arguments is considered as a bad practice in general, please assign this to a new variable